### PR TITLE
Ollama integration tile

### PR DESCRIPTION
**Pull request by AlistairLR112**
Adding tile for ollama integration - https://github.com/deepset-ai/haystack-core-integrations/tree/main/integrations/ollama

### CODE REVIEW

**Comment by bilgeyucel** [comment]
Hi @AlistairLR112, thank you for the PR! I'll review it as soon as possible 🙌

**Comment by AlistairLR112** [comment]
Wait until https://github.com/deepset-ai/haystack-core-integrations/pull/170 approved :)

File: integrations/ollama.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+### Table of Contents
+
+- [Introduction](#introduction)
+- [Installation](#installation)
+- [Usage](#usage)
+
+### Introduction
+
+You can use [Ollama Models](https://ollama.ai/library) in your Haystack 2.0 pipelines with the OllamaGenerator
```
Comment (anakin87) [inline integrations/ollama.md:35]:
```suggestion
You can use [Ollama Models](https://ollama.ai/library) in your Haystack 2.0 pipelines with the OllamaGenerator.
```

File: integrations/ollama.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+[Ollama](https://ollama.ai/) allows you to get up and running with large language models, locally
```
Comment (anakin87) [inline integrations/ollama.md:37]:
```suggestion
[Ollama](https://ollama.ai/) is a project focused on running Large Language Models locally. Internally it uses the quantized GGUF format by default. This means it is possible to run LLMs on standard machines (even without GPUs), without having to handle complex installation procedures.
```

File: integrations/ollama.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+```bash
+pip install ollama-haystack
+```
+
+### Usage
+
+You can use Ollama models in various ways:
+
+#### Generative Models (LLMs)
```
Comment (anakin87) [inline integrations/ollama.md:49]:
```suggestion
```
This part can be removed.

**Review by anakin87 — COMMENTED** [review]
I only left some minor comments.
@bilgeyucel please do the actual review (when you can)

File: integrations/ollama.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+---
+
+### Table of Contents
+
+- [Introduction](#introduction)
+- [Installation](#installation)
+- [Usage](#usage)
+  - [Example](#example)
+
+### Introduction
```
Comment (bilgeyucel) [inline integrations/ollama.md:34]: These titles should be h2 with `##` to be displayed in the table of contents
```suggestion
## Introduction
```

File: integrations/ollama.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+You can use [Ollama Models](https://ollama.ai/library) in your Haystack 2.0 pipelines with the OllamaGenerator.
+
+[Ollama](https://ollama.ai/) is a project focused on running Large Language Models locally. Internally it uses the quantized GGUF format by default. This means it is possible to run LLMs on standard machines (even without GPUs), without having to handle complex installation procedures.
+
+### Installation
```
Comment (bilgeyucel) [inline integrations/ollama.md:40]:
```suggestion
## Installation
```

File: integrations/ollama.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+```bash
+pip install ollama-haystack
+```
+
+### Usage
```
Comment (bilgeyucel) [inline integrations/ollama.md:46]:
```suggestion
## Usage
```

File: integrations/ollama.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+You can leverage Ollama models through the OllamaGenerator Component
+
+To use an Ollama model for text generation:
+
+1. Follow instructions on the [Ollama Github Page](https://github.com/jmorganca/ollama) to pull and serve your model of choice 
+2. Initialize an `OllamaGenerator` with the name of the model served in your Ollama instance and you can then use the `OllamaGenerator` instance in a question answering pipeline after the `PromptBuilder`.  
+
+
+#### Example
```
Comment (bilgeyucel) [inline integrations/ollama.md:56]:
```suggestion
### Example
```

File: integrations/ollama.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        Document(content="Super Mario was an important politician"),
+        Document(content="Mario owns several castles and uses them to conduct important political business"),
+        Document(
+            content="Super Mario was a successful military leader who fought off several invasion attempts by "
+            "his arch rival - Bowser"
+        ),
+    ]
+)
+
+query = "Who is Super Mario?"
```
Comment (bilgeyucel) [inline integrations/ollama.md:87]: Can you put this line `query = "Who is Super Mario?"` right before `response = pipe.run({"prompt_builder": {"query": query}, "retriever": {"query": query}})`? It makes the example easier to follow 👍

File: integrations/ollama.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    {{ document.content }}
+{% endfor %}
+
+Question: {{ query }}?
+"""
+pipe = Pipeline()
+
+pipe.add_component("retriever", InMemoryBM25Retriever(document_store=document_store))
+pipe.add_component("prompt_builder", PromptBuilder(template=template))
+pipe.add_component("llm", OllamaGenerator(model="orca-mini"))
```
Comment (bilgeyucel) [inline integrations/ollama.md:104]: Does OllamaGenerator require any `api_key` or `url`? If so, can you add that to the example with a template value like `OllamaGenerator(model="orca-mini", api_key="API_KEY")`

**Review by bilgeyucel — COMMENTED** [review]
Thank you @AlistairLR112! I left some minor comments. 

One more comment: can you change the logo something with a background (maybe white)? We're using a dark background for logos and the with transparent bg won't be visible

File: integrations/ollama.md (hunk 9 above)
Comment (AlistairLR112) [reply integrations/ollama.md:104]: It requires a URL that is already set as a default - the Ollama service defaults to running on http://localhost:11434/api/generate - can add that if you think it's needed

https://github.com/deepset-ai/haystack-core-integrations/blob/main/integrations/ollama/src/ollama_haystack/generator.py

**Comment by TuanaCelik** [comment]
Hi @AlistairLR112 and @anakin87 - thanks both for your contribution here! We will aim to announce this on Monday. @AlistairLR112 - please let me know if there's a twitter/LI account you'd like tagged. Also, if you have a demo or want to showcase this integration, we can help with that too.

**Review by TuanaCelik — APPROVED** [review]

**Comment by AlistairLR112** [comment]
Hey @TuanaCelik! Could you tag my linkedin here?
https://www.linkedin.com/in/alistairlr/

**Comment by sachinsachdeva** [comment]
@AlistairLR112 and @anakin87, looks like you guys had all the fun without me :) , 
hopefully we get a chance again .

**Pull request merged**